### PR TITLE
*: improve some links

### PR DESCRIPTION
**Pull request by breeswish**
1. Added links to SIG list and WG list.
2. Fixed Challenge Program broken link.
3. Modified link titles to be clear about external resources or in-this-repo resources.

Signed-off-by: Breezewish <me@breeswish.org>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,27 +13,27 @@ improving docs, improving code, giving talks, organizing meetups, etc.
...
 
 ### Communication
 
-The [communication](./communicating.md) page lists communication channels like
+The [communicating.md](./communicating.md) file lists communication channels like
 chat, social medias, etc.
 
 For more specific topics, join
-[TiDB Community](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE) slack channel and discuss with others.
+[TiDB Community slack channel](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE) and discuss with others.
```
Comment (zz-jason) [inline README.md]:
```suggestion
[TiDB Community slack workspace](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE) and discuss with others.
```

File: README.md [hunk 2]
```diff
@@ -43,21 +43,24 @@ TiDB has the following types of groups that are officially supported:
   module of TiDB. SIGs must have open and transparent proceedings. The purpose
   of a SIG is to own and develop a module of TiDB.
 
+  See [`special-interest-groups` directory](./special-interest-groups) for details and active SIGs.
```
Comment (zz-jason) [inline README.md]:
```suggestion
  See [`special-interest-groups`](./special-interest-groups/README.md) for details and active SIGs.
```

File: README.md [hunk 3]
```diff
@@ -43,21 +43,24 @@ TiDB has the following types of groups that are officially supported:
...
+
 * **Working Groups (WGs)** are temporary groups that are formed to address
   issues that cross SIG boundaries. Working groups do not own any code or other
   long term artifacts. Working groups can report back and act through involved
   SIGs.
 
+  See [`working-groups` directory](./working-groups) for details and active WGs.
```
Comment (zz-jason) [inline README.md]:
```suggestion
  See [`working-groups`](./working-groups/README.md) for details and active WGs.
```

File: README.md [hunk 4]
```diff
@@ -43,21 +43,24 @@ TiDB has the following types of groups that are officially supported:
...
+
 ### How to contribute
 
-Contributions are welcomed and greatly appreciated. See [How to
-contribute](./contributors/README.md) for details on setting up development
-environment, contribution workflow and submitting patches, etc.
+Contributions are welcomed and greatly appreciated.
+
+See [`contributors` directory](./contributors/README.md) for details.
```
Comment (zz-jason) [inline README.md]:
```suggestion
See [`contributors`](./contributors/README.md) for details.
```

File: README.md [hunk 5]
```diff
@@ -43,21 +43,24 @@ TiDB has the following types of groups that are officially supported:
...
 
 ## Learning Resources
 
-Learning resources are collected in the [learning
-resources](./learning-resources/README.md) page. Here you can find all the
+Learning resources are collected in the [`learning-resources` directory](./learning-resources/README.md). Here you can find all the
```
Comment (zz-jason) [inline README.md]:
```suggestion
Learning resources are collected in the [`learning-resources`](./learning-resources/README.md). Here you can find all the
```

File: README.md [hunk 6]
```diff
@@ -68,7 +71,7 @@ TiDB architecture though the following blogs:
 
 ## RFC
 
-See [RFC](./rfc/README.md) for RFCs for details.
+See [`rfc` directory](./rfc/README.md).
```
Comment (zz-jason) [inline README.md]:
```suggestion
See [`rfc`](./rfc/README.md).
```

**Comment by breeswish** [comment]
@zz-jason I think for paragraphs like `see xxx`, `xxx` should be the description of the **location**, instead of the description of the **content**, because the description of the content is already described by the title.

Consider the following examples. In example 1, it doesn't give any information about where to see the details.

## Example 1

### TiDB Roadmap

See TiDB Roadmap for details.

## Example 2

### TiDB Roadmap

See docs/roadmap.md for details.

File: README.md [hunk 7]
```diff
@@ -13,27 +13,27 @@ improving docs, improving code, giving talks, organizing meetups, etc.
 
 ![TiDB Community Architecture](./media/architecture.svg)
 
-See [TiDB Community Architecture](./architecture/README.md) for more details.
+See [`architecture` directory](./architecture/README.md) for details.
```
Comment (zz-jason) [inline README.md]:
```suggestion
See [`architecture`](./architecture/README.md) for details.
```

File: README.md [hunk 8]
```diff
@@ -43,21 +43,24 @@ TiDB has the following types of groups that are officially supported:
   module of TiDB. SIGs must have open and transparent proceedings. The purpose
   of a SIG is to own and develop a module of TiDB.
 
+  See [`special-interest-groups` directory](./special-interest-groups/README.md) for details and active SIGs.
```
Comment (zz-jason) [inline README.md]:
```suggestion
  See [`special-interest-groups`](./special-interest-groups/README.md) for details and active SIGs.
```

**Pull request merged**